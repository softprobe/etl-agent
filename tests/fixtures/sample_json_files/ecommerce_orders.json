[
  {
    "order_id": "ORD-2024-001",
    "customer_id": 12345,
    "customer_email": "john.doe@example.com",
    "order_date": "2024-01-15T10:30:00Z",
    "status": "completed",
    "total_amount": 156.78,
    "currency": "USD",
    "shipping_address": {
      "street": "123 Main St",
      "city": "San Francisco",
      "state": "CA",
      "zip_code": "94105",
      "country": "US"
    },
    "items": [
      {
        "product_id": "PROD-001",
        "product_name": "Wireless Headphones",
        "category": "Electronics",
        "quantity": 2,
        "unit_price": 49.99,
        "total_price": 99.98
      },
      {
        "product_id": "PROD-002", 
        "product_name": "Phone Case",
        "category": "Accessories",
        "quantity": 1,
        "unit_price": 19.99,
        "total_price": 19.99
      }
    ],
    "payment": {
      "method": "credit_card",
      "last_four": "4567",
      "transaction_id": "TXN-ABC123"
    },
    "shipping": {
      "method": "standard",
      "cost": 9.99,
      "estimated_delivery": "2024-01-18T00:00:00Z"
    }
  },
  {
    "order_id": "ORD-2024-002",
    "customer_id": 67890,
    "customer_email": "jane.smith@example.com",
    "order_date": "2024-01-16T14:22:00Z",
    "status": "pending",
    "total_amount": 89.99,
    "currency": "USD",
    "shipping_address": {
      "street": "456 Oak Ave",
      "city": "New York",
      "state": "NY", 
      "zip_code": "10001",
      "country": "US"
    },
    "items": [
      {
        "product_id": "PROD-003",
        "product_name": "Coffee Mug",
        "category": "Kitchen",
        "quantity": 3,
        "unit_price": 12.99,
        "total_price": 38.97
      },
      {
        "product_id": "PROD-004",
        "product_name": "Notebook Set",
        "category": "Office",
        "quantity": 2,
        "unit_price": 15.99,
        "total_price": 31.98
      }
    ],
    "payment": {
      "method": "paypal",
      "transaction_id": "PP-XYZ789"
    },
    "shipping": {
      "method": "express",
      "cost": 19.04,
      "estimated_delivery": "2024-01-17T00:00:00Z"
    }
  }
]