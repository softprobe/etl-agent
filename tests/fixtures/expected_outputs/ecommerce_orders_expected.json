{
  "business_entities": [
    {
      "name": "Order",
      "type": "transaction",
      "primary_key": "order_id",
      "description": "E-commerce order transaction"
    },
    {
      "name": "Customer", 
      "type": "entity",
      "primary_key": "customer_id",
      "description": "Customer information"
    },
    {
      "name": "OrderItem",
      "type": "line_item", 
      "primary_key": null,
      "parent_entity": "Order",
      "description": "Individual items within an order"
    },
    {
      "name": "Product",
      "type": "catalog",
      "primary_key": "product_id",
      "description": "Product catalog information"
    }
  ],
  "suggested_schema": {
    "orders": {
      "order_id": {"type": "STRING", "nullable": false, "description": "Unique order identifier"},
      "customer_id": {"type": "INTEGER", "nullable": false, "description": "Customer identifier"},
      "customer_email": {"type": "STRING", "nullable": false, "description": "Customer email address"},
      "order_date": {"type": "TIMESTAMP", "nullable": false, "description": "Order creation timestamp"},
      "status": {"type": "STRING", "nullable": false, "description": "Order status"},
      "total_amount": {"type": "FLOAT", "nullable": false, "description": "Total order amount"},
      "currency": {"type": "STRING", "nullable": false, "description": "Currency code"},
      "shipping_street": {"type": "STRING", "nullable": true, "description": "Shipping address street"},
      "shipping_city": {"type": "STRING", "nullable": true, "description": "Shipping address city"},
      "shipping_state": {"type": "STRING", "nullable": true, "description": "Shipping address state"},
      "shipping_zip": {"type": "STRING", "nullable": true, "description": "Shipping address zip code"},
      "shipping_country": {"type": "STRING", "nullable": true, "description": "Shipping address country"},
      "payment_method": {"type": "STRING", "nullable": true, "description": "Payment method used"},
      "transaction_id": {"type": "STRING", "nullable": true, "description": "Payment transaction ID"},
      "shipping_method": {"type": "STRING", "nullable": true, "description": "Shipping method"},
      "shipping_cost": {"type": "FLOAT", "nullable": true, "description": "Shipping cost"}
    },
    "order_items": {
      "order_id": {"type": "STRING", "nullable": false, "description": "Reference to order"},
      "product_id": {"type": "STRING", "nullable": false, "description": "Product identifier"},
      "product_name": {"type": "STRING", "nullable": false, "description": "Product name"},
      "category": {"type": "STRING", "nullable": true, "description": "Product category"},
      "quantity": {"type": "INTEGER", "nullable": false, "description": "Item quantity"},
      "unit_price": {"type": "FLOAT", "nullable": false, "description": "Price per unit"},
      "total_price": {"type": "FLOAT", "nullable": false, "description": "Total price for this item"}
    }
  },
  "confidence_scores": {
    "schema_detection": 0.95,
    "business_entity_detection": 0.90,
    "data_type_inference": 0.92,
    "relationship_detection": 0.85
  },
  "potential_issues": [
    {
      "type": "nested_structure",
      "description": "Nested shipping_address and items arrays need flattening",
      "severity": "medium",
      "suggested_solution": "Create separate tables for order_items and flatten shipping address fields"
    },
    {
      "type": "optional_fields",
      "description": "Some payment fields may be missing depending on payment method", 
      "severity": "low",
      "suggested_solution": "Mark payment-specific fields as nullable"
    }
  ]
}